#! /bin/sh /usr/share/dpatch/dpatch-run
## lmce_filldata.dpatch by Daniel Kristjansson <danielk@cuymedia.net>

@DPATCH@
diff -urNad mythtv-0.20+fixes15192~/programs/mythfilldatabase/filldata.cpp mythtv-0.20+fixes15192/programs/mythfilldatabase/filldata.cpp
--- blah/programs/mythfilldatabase/filldata.cpp	(revision 15983)
+++ blah/programs/mythfilldatabase/filldata.cpp	(working copy)
@@ -47,6 +47,7 @@
 
 static QString SetupIconCacheDirectory();
 
+bool contact_backend = true;
 bool interactive = false;
 bool channel_preset = false;
 bool non_us_updating = false;
@@ -3512,6 +3513,10 @@
             cout << "###\n";
             interactive = true;
         }
+        else if (!strcmp(a.argv()[argpos], "--ignore-backend"))
+        {
+            contact_backend = false;
+        }
         else if (!strcmp(a.argv()[argpos], "--preset"))
         {
             // For using channel preset values instead of channel numbers.
@@ -3888,6 +3893,11 @@
             cout << "   Use '-v help' for level info\n";
             cout << "\n";
 
+            cout << "--ignore-backend\n";
+            cout << "   LMCE specific param that ignores the backend,\n";
+            cout << "   so we can issue mythbackend --resched separately.\n";
+            cout << "\n";
+
 #if 0
             cout << "--dd-grab-all\n";
             cout << "   The DataDirect grabber will grab all available data\n";
@@ -4274,17 +4284,21 @@
         DataDirectProcessor::FixProgramIDs();
     }
 
-    VERBOSE(VB_IMPORTANT, "\n"
+    if (contact_backend)
+    {
+        VERBOSE(
+            VB_IMPORTANT, "\n"
             "===============================================================\n"
             "| Attempting to contact the master backend for rescheduling.  |\n"
             "| If the master is not running, rescheduling will happen when |\n"
             "| the master backend is restarted.                            |\n"
             "===============================================================");
 
-    if (grab_data || mark_repeats)
-        ScheduledRecording::signalChange(-1);
+        if (grab_data || mark_repeats)
+            ScheduledRecording::signalChange(-1);
 
-    RemoteSendMessage("CLEAR_SETTINGS_CACHE");
+        RemoteSendMessage("CLEAR_SETTINGS_CACHE");
+    }
 
     delete gContext;
 
